jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java

```diff
@@ -176,34 +175,31 @@ public ObjectNode buildFido2AssertionStartResponse(String userName, String keyHa
     public JsonNode finishAuthentication(String userName, String authenticateResponseString) {
         AuthenticateResponse authenticateResponse = parseAuthenticateResponse(authenticateResponseString);
 
-        ObjectNode params = buildFido2AuthenticationVerifyResponse(userName, authenticateResponseString, authenticateResponse);
+        AssertionResult params = buildFido2AuthenticationVerifyResponse(userName, authenticateResponseString, authenticateResponse);
 
-        ObjectNode result = assertionService.verify(params);
+        AssertionResultResponse assertionResultResponse = assertionService.verify(params);
+        ObjectNode result = (ObjectNode)CommonUtilService.toJsonNode(assertionResultResponse);
 
         result.put("status", "success");
         result.put("challenge", authenticateResponse.getClientData().getChallenge());
 
         return result;
     }
 
-    public ObjectNode buildFido2AuthenticationVerifyResponse(String userName, String authenticateResponseString, AuthenticateResponse authenticateResponse) {
+    public AssertionResult buildFido2AuthenticationVerifyResponse(String userName, String authenticateResponseString, AuthenticateResponse authenticateResponse) {
         if (!ArrayUtils.contains(RawAuthenticationService.SUPPORTED_AUTHENTICATE_TYPES, authenticateResponse.getClientData().getTyp())) {
             throw errorResponseFactory.badRequestException(AssertionErrorResponseType.UNSUPPORTED_AUTHENTICATION_TYPE, "Invalid options attestation request type");
         }
 
         boolean oneStep = StringHelper.isEmpty(userName);
 
-        ObjectNode params = dataMapperService.createObjectNode();
+        AssertionResult assertionResult = new AssertionResult();
         // Add all required parameters from request to allow process U2F request 
-        params.put(CommonVerifiers.SUPER_GLUU_REQUEST, true);
-        params.put(CommonVerifiers.SUPER_GLUU_MODE, oneStep ? SuperGluuMode.ONE_STEP.getMode() : SuperGluuMode.TWO_STEP.getMode());
-
-        // Manadatory parameter
-        params.put("type", "public-key");
-
-        params.put("id", authenticateResponse.getKeyHandle());
-
-        params.put("rawId", authenticateResponseString);
+        assertionResult.setSuperGluuRequest(true);
+        assertionResult.setSuperGluuRequestMode(oneStep ? SuperGluuMode.ONE_STEP.getMode() : SuperGluuMode.TWO_STEP.getMode());
+        assertionResult.setType("public-key");
```

Comment:
Constructor should have type set by default.

If you actually want to set type, then it should use instance of PublicKeyCredentialType

Reply:
done